### PR TITLE
[runtime-security] add test and tool to test and debug process resolver

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR add unit and functional tests testing process resolver. It also add a security-agent command line to trigger a process resolver cache dump that can be rendered as a dot graph.

### CODE REVIEW

File: pkg/security/api/api.proto [hunk 1]
```diff
@@ -1,15 +1,24 @@
 syntax = "proto3";
 
+option go_package = "pkg/security/api";
+
 package api;
 
-message GetParams{}
+message GetEventParams{}
```
Comment (Gui774ume) [inline pkg/security/api/api.proto:7]: nit: `GetEvents`

File: pkg/security/api/api.proto [hunk 2]
```diff
@@ -1,15 +1,24 @@
...
 
+message DumpProcessCacheParams{}
+
+message SecurityDumpProcessCacheMessage {
+    string Filename = 1;
+}
+
 service SecurityModule {
-    rpc GetEvents(GetParams) returns (stream SecurityEventMessage) {}
+    rpc GetEvents(GetEventParams) returns (stream SecurityEventMessage) {}
```
Comment (Gui774ume) [inline pkg/security/api/api.proto:22]: nit: `GetEventsParams`

**Review by lebauce — APPROVED** [review]

**Review by Gui774ume — APPROVED** [review]
2 nits but feel free to ignore. lgtm :+1:

File: pkg/security/api/api.proto (hunk 1 above)
Comment (safchain) [reply pkg/security/api/api.proto:7]: That is the parameter of the rpc method, I renamed it as now we have 2 methods with two kind of params

**Pull request merged**